trunk/includes/pardot-forms-widget-class.php

```diff
@@ -281,13 +281,15 @@ function update( $new_instance, $old_instance ) {
 		/**
 		 * Default the 'form_id' to a non-selected form ID, i.e. '0'.
 		 */
-		$instance = array( 'form_id' => 0 );
+		$instance = ['form_id' => 0];
 
 		/**
-		 * If the new instance has 'form_id' then capture it's value before returning.
+		 * If the new instance has 'form_id' then capture its value before returning.
 		 */
-		if ( isset( $new_instance['form_id'] ) )
+		if ( isset( $new_instance['form_id'] ) ) {
 			$instance['form_id'] = $new_instance['form_id'];
+			Pardot_Plugin::delete_form_html_transient($new_instance['form_id']);
```

Comment:
This deletes the existing cached HTML for the form upon selection when adding a widget.  The next time the form is loaded (likely when the user previews the page), the new HTML for this form will be pulled and cached.  This eliminates the scenario where the HTML for a form is out of a date (like using an old domain) and we continue using the old HTML.  Previously, users would need to click the clear cache button every time they wanted to pull a new domain for a form that was previously loaded.  Now, selecting a form will refresh the HTML for that particular form.  The hope is that this will limit how often users have to click the clear cache button.